p2p/src/sync/peer_v2.rs

```diff
@@ -405,16 +406,9 @@ where
         }
 
         if self.chainstate_handle.is_initial_block_download().await? {
-            // TODO: in the protocol v2 we might want to allow peers to ask for headers even if
-            // the node is in IBD (e.g. based on some kind of system of permissions). ATM it's
-            // not clear whether it's a good idea, so it makes sense to first check whether bitcoin
-            // does something like that.
-            // TODO: in the protocol v2 we should not silently ignore header requests; instead,
-            // we should communicate our best block (id/height/header?) from the start, so that
-            // the peer just knows that we don't have better blocks and doesn't ask us in the
-            // first place.
-            // See the issue #1110.
             log::debug!("[peer id = {}] Ignoring headers request because the node is in initial block download", self.id());
+            // Respond with an empty list to avoid being marked as stalled
+            self.send_headers(HeaderList::new(Vec::new()))?;
```

Comment:
There's an open question here whether this has a semi-predicate problem... do we need to distinguish between IBD and not? By using this kind of empty response, we're doing the same "special value semantics" that can only lead to problems. Isn't it more reasonable to wrap this in an enum stating that it's empty because "I'm in IBD"?

Reply:
The open question is whether there's some benefit we're not noticing.